### PR TITLE
Add support for boleto

### PR DESCRIPTION
**Pull request by eurias-stripe**
## Summary
Add support for boleto

## Motivation
LPM Sprint

## Testing
Simulator Screen Recording - iPhone 14 Pro - 2023-09-11 at 12 19 07

Added tests
Verified manually

## Changelog
* [Added] Support for Boleto with PaymentIntents.

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+Boleto.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    func makeBoleto() -> FormElement {
+        let contactInfoSection = makeContactInformationSection(
+            nameRequiredByPaymentMethod: true,
+            emailRequiredByPaymentMethod: true,
+            phoneRequiredByPaymentMethod: false
+        )
+        let taxIdElement = TextFieldElement(
+            configuration: IDNumberTextFieldConfiguration(
+                type: .BR_CPF_CNPJ,
+                label: STPLocalizedString("CPF/CPNJ", "Label for CPF/CPNJ (Brazil tax ID) field")
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+Boleto.swift:23]: nit: preferred pattern is to add to `String+Localized.swift` in the PaymentSheet module

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift [hunk 2]
```diff
@@ -168,6 +168,8 @@ extension STPPaymentMethodType {
                     return .pm_type_alipay
                 case .OXXO:
                     return .pm_type_oxxo
+                case .boleto:
+                    return .pm_type_boleto
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:172]: Don't forget to check dark mode - might need to add .boleto to `iconRequiresTinting` above

**Review by yuki-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift (hunk 2 above)
Comment (eurias-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:179]: Good catch, this icon doesn't look good in dark mode and the tinting makes it worse, I guess I need to edit the icon
No tinting:
image
Tinting:
image

Comment (eurias-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:179]: Fixed now
Simulator Screenshot - iPhone 14 Pro - 2023-09-14 at 09 29 50

**Review by yuki-stripe — APPROVED** [review]

**Pull request merged**